src/components/report-viewer/report-viewer.tsx

```diff
@@ -34,68 +25,50 @@ export default function ReportViewer({ report, reportType }: ReportViewerProps)
     const [selectedReportId, setSelectedReportId] = useState(report?.id);
     const [selectedReportType, setSelectedReportType] = useState<ReportType>();
 
-    const reportTreeData = useRef<Record<string, ReportTreeType>>({});
-    const treeView = useRef<ReactElement>();
+    const reportTree = useMemo(() => mapReportsTree(report), [report]);
+    const reportTreeMap = useRef<Record<string, ReportTree>>({});
 
-    /**
-     * Build the tree view (left pane) creating all ReportItem from json data
-     * @type {Function}
-     */
-    const initializeTreeDataAndComponent = useCallback((report: ReportTreeType) => {
-        reportTreeData.current[report.id] = report;
-        return (
-            <ReportItem
-                labelText={report.message}
-                labelIconColor={report.severity.colorName}
-                key={report.id}
-                sx={styles.treeItem}
-                nodeId={report.id}
-            >
-                {report.subReports.map((value: ReportTreeType) => initializeTreeDataAndComponent(value))}
-            </ReportItem>
-        );
+    const mapReportsById = useCallback((item: ReportTree) => {
+        reportTreeMap.current[item.id] = item;
```

Comment:
should maybe throw away the old map and assign a new empty object to reportTreeMap.current ?